lib/rubycritic/analysers/coverage.rb

```diff
@@ -87,7 +87,7 @@ def stored_data
       # given time
       def synchronize_resultset
         # make it reentrant
-        return yield if @resultset_locked == true
+        return yield if defined?(@resultset_locked) && @resultset_locked == true
```

Comment:
@etagwerker mind taking a look the first two commits in this PR? I've noticed them when preparing a release :)

Reply:
@Onumis Sure, I'll take a look